### PR TITLE
QMake: Linux installation improvements

### PR DESCRIPTION
**Pull request by dyfer**
This PR aims to add Linux desktop resource installation for feature parity with the `meson` build. Changes include:
- creation of `.desktop` and `.metainfo` files
- copying and renaming icons
- adding `install` option to the `build` script
- updating Linux build instructions
- ~~**possibly unrelated change**: on my Ubuntu 20.04 system jack was not found with pkg-config. I tried to fix it, but did not manage to do it (I've reinstalled `libjack-jackd2-dev` and `pkg-config` but that did not fix it); I've then added a flag in our `jacktrip.pro` to check if `jack` flags are available from `pkg-config`, and if not, use the previously hardcoded flags (`-ljack`), which works on my system. Let me know in case this change should be removed or put in a separate PR.~~ I reverted this change - it's probably better if we fail when jack is not found (*)

With these changes, qmake build gains most of the improvements from #361: Manpage, Icons, and Metainfo / Appstream data. If these changes look good, IMO we could release 1.4.0 without moving to meson (currently held up by an [issues with static build](https://github.com/jacktrip/jacktrip/issues/361#issuecomment-889516395)) and move to meson when ready. 

Previously Linux qmake instructions provided a rather manual installation instructions (`cp` and `chmod`...). We have however, as it turns out, installation feature in qmake (`(sudo) make release-install`). The updated instructions (and the build script) take advantage of that.

What do you all think?

EDIT:
(*) My problem with pkg-config was due to having a homebrew install of pkg-config (yes, on linux) which was listing only packages installed with homebrew. `brew unlink pkg-config` fixed it.

### CODE REVIEW

File: docs/Build/Linux.md [hunk 1]
```diff
@@ -23,83 +23,112 @@ dnf install "pkgconfig(jack)" alsa-lib-devel git help2man
...
+You can pass configuration options to qmake with `-config <option>`, e.g. `qmake
+-config nogui ../jacktrip.pro`. Relevant options include:
+- `-config rtaudio` - build with rtaudio support
+- `-config nogui` - build without the GUI
+- `-config static` - build with static libraries
+- `-config weakjack` - build using weak linking of jack
+- `-config nojack` - build without jack support
+
 To build using QtCreator:
   * Open jacktrip.pro using QtCreator
```
Comment (ntonnaett) [inline docs/Build/Linux.md:111]: Not your change. But this doesn't print correctly. Maybe you want to fix this too.

File: jacktrip.pro [hunk 2]
```diff
@@ -273,3 +283,47 @@ rtaudio {
...
+    appdata = $$cat($${PWD}/linux/org.jacktrip.JackTrip.metainfo.xml.in, blob)
+    appdata = $$replace(appdata, @appid@, $${application_id})
+    write_file($${OUT_PWD}/$${application_id}.metainfo.xml, appdata)
+
+    metainfo.files = $${OUT_PWD}/$${application_id}.metainfo.xml
+    metainfo.path = $${PREFIX}/share/metainfo
+
+    desktop_conf = $$cat($${PWD}/linux/org.jacktrip.JackTrip.desktop.in, blob)
+    desktop_conf = $$replace(desktop_conf, @icon@, $${application_id})
+    desktop_conf = $$replace(desktop_conf, @wmclass@, lower($${application_id}))
```
[the remark this answers is not in the document]
Comment (dyfer) [reply jacktrip.pro:313]: thanks for catching this!

File: build [hunk 3]
```diff
@@ -116,3 +122,6 @@ if [[ $clean == 1 ]]; then
 fi
 $QCMD -spec $QSPEC $CONFIG ../jacktrip.pro $UNKNOWN_OPTIONS
 $MCMD release
+if [[ $install == 1 ]]; then
+  sudo $MCMD release-install
```
[the remark this answers is not in the document]
Comment (dyfer) [reply build:126]: Di you mean something like:
```
echo "We need elevated privileges to install JackTrip in the system location"
sudo (...)
```
?

File: docs/Build/Linux.md (hunk 1 above)
Comment (dyfer) [reply docs/Build/Linux.md:111]: Thanks for catching. I think my list before renders wrong too - MkDocs seems to require an empty line before a list (my local Markdown preview doesn't)

File: build (hunk 3 above)
[the remark this answers is not in the document]
Comment (ntonnaett) [reply build:126]: Yes!

**Comment by ntonnaett** [comment]
It also looked that the renaming of the icons didn't work. Could you look into this?

**Comment by dyfer** [comment]
> It also looked that the renaming of the icons didn't work. Could you look into this?

Thanks for catching that. It worked on subsequent installs from the same dir, but not on the first one. Looking into fixing it.

**Comment by dyfer** [comment]
Well, that's weird. Seems like install file needs to exist at the time of running qmake, and I had problems moving renaming of icons (and creating man page) to the build step. So instead I've used the `.extra` argument to directly run the copy (or help2man) command and output the file in the destination install directory. Let me know if this is an OK solution.

**Comment by dyfer** [comment]
I've found a better fix for my original problem - there's a flag that can be set that disables check for non-existing files. Now the install should work properly.

**Comment by ntonnaett** [comment]
I was already wondering why your pkgconfig fails. I think it's better to fail if it doesn't work.

File: jacktrip.pro (hunk 2 above)
[the remark this answers is not in the document]
Comment (ntonnaett) [reply jacktrip.pro]: This is still happening.

[the remark this answers is not in the document]
Comment (ntonnaett) [reply jacktrip.pro]: Sorry. Nevermind.

**Review by ntonnaett — APPROVED** [review]

**Pull request merged**